client/src/main/java/org/evosuite/ga/metaheuristics/TestSuiteAdapter.java

```diff
@@ -312,28 +325,28 @@ final protected void calculateFitness(TestSuiteChromosome c)
     }
 
     @Override
-    final protected void calculateFitnessAndSortPopulation() throws UnsupportedOperationException {
-        throw new UnsupportedOperationException("not implemented");
+    final protected void calculateFitnessAndSortPopulation(){
+        algorithm.calculateFitnessAndSortPopulation();
     }
 
     @Override
-    final public int getPopulationSize() throws UnsupportedOperationException {
-        throw new UnsupportedOperationException("not implemented");
+    final public int getPopulationSize() {
+        return algorithm.getPopulationSize();
     }
 
     @Override
-    final protected List<TestSuiteChromosome> elitism() throws UnsupportedOperationException {
-        throw new UnsupportedOperationException("not implemented");
+    final protected List<TestSuiteChromosome> elitism() {
+        return algorithm.elitism().stream().map(TestChromosome::toSuite).collect(toList());
```

Comment:
Question: this gives you a list of `TestSuiteChromosomes`, and every such chromosome contains exactly one test case, right? Wouldn't it make more sense to return a list containing just one `TestSuiteChromosome`, which in turn contains all the test cases? This approach seems to be more in line with `getBestIndividuals()` in `MOSATestSuiteAdapter`, `MIOTestSuiteAdapter` and `LIPSTestSuiteAdapter`. I suppose this would be easy to implement, too, since we already have a `TestChromosomeCollector`.

Reply:
I think both approaches are fine.

I went with the List of test suites, because before the refactoring, this function would have returned a list of tests.

If you think we should return a List only containing one test suite, i am fine with it.

Reply:
OK, then let's keep your implementation :smile: